### PR TITLE
Deadline: reworked pools assignment

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Previous solution was nonconsistent, expected values for pools in multiple places in Setting.

## Description
New collector and validator was introduced. Collector adds pool names from Setting (merged and removed separate pool settings in hosts sections) if not already present on a instance (as Maya has it, potentially others with New Publisher).

Validator checks if name of pools on the instances matches to current values on DL (does a query of DL).

## Additional info
Refactored `_requests_get` and `requests_post` wrappers into `openpype.lib.connections` - name up for commenting/changing.

## Testing notes:
1. configure `project_settings/deadline/publish/CollectDeadlinePools`
1.a - insert non existent values
1.b - insert values matching situation on DL
3. publish on DL and check pools

### CODE REVIEW

**Comment by m-u-r-p-h-y** [comment]
- Maya render instance (and all render instances in other hosts) should check the settings and prefill the pools and groups from the settings
- Pools and Groups should not be a drop-down, but text fields as we can combine multiple groups to fine-tune specific scene requirements. Ex. I need to submit my heavy scene to group Maya, Ram256 and GPU to render on machines with Maya installed, with more than 256 GB RAM and GPU card)




I would suggest to add pools and group to Nuke write instance (the same as we have in Maya)

image

**Comment by m-u-r-p-h-y** [comment]
there is also an issue in collect render layers (I'm not sure if it is connected to this PR though)

```
// pyblish.CollectMayaRender : collecting layer: rs_Main // 
// Error: pyblish.plugin : Traceback (most recent call last):
  File "D:\REPO\OpenPype\.venv\lib\site-packages\pyblish\plugin.py", line 522, in __explicit_process
    runner(*args)
  File "D:\REPO\OpenPype\openpype\hosts\maya\plugins\publish\collect_render.py", line 391, in process
KeyError: 'SecondaryPool'
Traceback (most recent call last):
  File "D:\REPO\OpenPype\.venv\lib\site-packages\pyblish\plugin.py", line 522, in __explicit_process
    runner(*args)
  File "<string>", line 391, in process
KeyError: 'SecondaryPool' // 
b'Processing Context Label'
```

File: openpype/modules/deadline/deadline_module.py [hunk 1]
```diff
@@ -32,3 +44,37 @@ def get_plugin_paths(self):
...
+            webservice (str): Server url.
+            log (Logger)
+        Returns:
+            list: Pools.
+        Throws:
+            RuntimeError: If deadline webservice is unreachable.
+
+        """
+        if not log:
+            from openpype.lib import PypeLogger
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/deadline_module.py:62]: Move the import to the top of file

File: openpype/modules/deadline/deadline_module.py [hunk 2]
```diff
@@ -32,3 +44,37 @@ def get_plugin_paths(self):
...
+
+            log = PypeLogger().get_logger(__name__)
```
Comment (iLLiCiTiT) [inline openpype/modules/deadline/deadline_module.py:64]: Initialization of `PypeLogger` is not needed.
```suggestion
            log = PypeLogger.get_logger(__name__)
```

**Comment by kalisp** [comment]
`collect_render` issue was resolved.

**Review by m-u-r-p-h-y — APPROVED** [review]
project pool is correctly propagated to render instance 

image

**Comment by BigRoy** [comment]
I'd expect users might want to set the Pool per project and per application, so that e.g. Maya runs in a different pool than Nuke or Harmony. Thoughts?

**Comment by m-u-r-p-h-y** [comment]
You are correct. I've already created issue #3057  to do it outside of this PR.

**Comment by kalisp** [comment]
@BigRoy I thought that pools are per project and groups would be per host.
But maybe we should add there profile for host, task name?

**Pull request merged**